TemporaryProjects/StartPageExtender.cs

```diff
@@ -77,6 +77,11 @@ private void TrackFrame(IVsWindowFrame frame)
                 if (ErrorHandler.Succeeded(frame.GetGuidProperty((int)__VSFPROPID.VSFPROPID_GuidPersistenceSlot, out var guid)) &&
                     guid == VSConstants.StandardToolWindows.StartPage &&
                     ErrorHandler.Succeeded(frame.GetProperty((int)__VSFPROPID.VSFPROPID_DocView, out var windowPane)) &&
+                    // The type of windowPane is Microsoft.VisualStudio.Shell.ToolWindowPane (which inherits from WindowPane),
+                    // but if we tried to cast it, the cast would fail at runtime in VS 2017 where the real type comes from
+                    // Microsoft.VisualStudio.Shell.15.0, whereas we're referencing
+                    // Microsoft.VisualStudio.Shell.14.0.
+                    // Both can't be referenced at the same time so we have to use reflection as a workaround.
                     windowPane?.GetType().GetProperty(nameof(WindowPane.Content)).GetValue(windowPane) is IVsUIWpfElement wpfElement &&
                     ErrorHandler.Succeeded(wpfElement.GetFrameworkElement(out var element)) &&
                     element is FrameworkElement frameworkElement &&
```

Comment:
Shall we see how this looks after adding a `TemporaryProjects.15.0` project?

Reply:
I see what you mean about  big and unreadable. I'm struggling a bit to understand what's going on. 😉 

I think it could be simplified a little using `dynamic`.
```            ErrorHandler.ThrowOnFailure(windowFrame.GetProperty((int)__VSFPROPID.VSFPROPID_DocView, out dynamic windowPane));
var wpfElement = windowPane.Content as IVsUIWpfElement;
```

Not that we'd need this if we have a `TemporaryProjects.15.0`. Just fun looking at different ways of doing things. 😄

Reply:
I think I tried using dynamic but it didn't work.

Reply:
Also, I have experience of getting a 1-2 second delay in my app when dynamic is first used.

Reply:
Switched to dynamic in fa0e100fd5e83a019d01bb40ff2ce61e05ee7782